cache/disk/options.go

```diff
@@ -42,14 +42,15 @@ func WithMaxBlobSize(size int64) Option {
 	}
 }
 
-func WithProxyBackend(proxy cache.Proxy) Option {
+func WithProxyBackend(proxy cache.Proxy, maxProxyBlobSize int64) Option {
```

Comment:
I think it would probably be nicer to add a `WithMaxProxyBlobSize` function instead of modifying `WithProxyBackend`.